back/app/services/idea_custom_fields_service.rb

```diff
@@ -141,6 +141,58 @@ def check_form_structure(fields, errors)
     end
   end
 
+  def duplicate_all_fields
+    fields = all_fields
+    logic_id_map = { survey_end: 'survey_end' }
+    copied_fields = fields.map do |field|
+      # Duplicate fields with a new id
+      copied_field = field.dup
+      copied_field.id = SecureRandom.uuid
+      logic_id_map[field.id] = copied_field.id
+
+      # Duplicate options with an id and temp_id
+      copied_options = field.options.map do |option|
+        copied_option = option.dup
+        copied_option.id = SecureRandom.uuid
+        copied_option.temp_id = "TEMP-ID-#{SecureRandom.uuid}"
+        logic_id_map[option.id] = copied_option.temp_id
+        copied_option
+      end
+      copied_field.options = copied_options
+
+      # Duplicate and persist map config if it is a point field
+      if copied_field.input_type == 'point'
+        original_map_config = CustomMaps::MapConfig.find(field.map_config.id)
```

Comment:
This is what fails if no `map_config` has been created for the `custom_field` (i.e. the map for the question was never configured)

Reply:
Now fixed by your change adding `&& field.map_config`